src/services/coretime/CoretimeService.ts

```diff
@@ -0,0 +1,613 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+// Copyright 2017-2024 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import { ApiDecoration, QueryableModuleStorage } from '@polkadot/api/types';
+import { Option, StorageKey, U16, U32, Vec } from '@polkadot/types';
+import { BlockHash, ParaId } from '@polkadot/types/interfaces';
+import {
+	PalletBrokerConfigRecord,
+	PalletBrokerLeaseRecordItem,
+	PalletBrokerPotentialRenewalId,
+	PalletBrokerPotentialRenewalRecord,
+	PalletBrokerRegionId,
+	PalletBrokerRegionRecord,
+	PalletBrokerSaleInfoRecord,
+	PalletBrokerScheduleItem,
+	PalletBrokerStatusRecord,
+	PolkadotRuntimeParachainsAssignerCoretimeCoreDescriptor,
+	PolkadotRuntimeParachainsParasParaLifecycle,
+} from '@polkadot/types/lookup';
+import { BN } from '@polkadot/util';
+
+import {
+	ICoretimeChainInfo,
+	ICoretimeCores,
+	ICoretimeLeases,
+	ICoretimeRegions,
+	ICoretimeRelayInfo,
+	ICoretimeRenewals,
+	ICoretimeReservations,
+	LeaseWithCore,
+	TConfigInfo,
+	TCoreDescriptor,
+	TLeaseInfo,
+	TParaLifecycle,
+	TPotentialRenewalInfo,
+	TRegionInfo,
+	TReservationInfo,
+	TSaleInfo,
+	TStatusInfo,
+	TWorkloadInfo,
+	TWorkplanInfo,
+} from '../../types/responses';
+import { AbstractService } from '../AbstractService';
+import {
+	extractConfigInfo,
+	extractCoreDescriptorInfo,
+	extractLeaseInfo,
+	extractParachainLifecycleInfo,
+	extractPotentialRenewalInfo,
+	extractRegionInfo,
+	extractReservationInfo,
+	extractSaleInfo,
+	extractStatusInfo,
+	extractWorkloadInfo,
+	extractWorkplanInfo,
+	sortByCore,
+} from './util';
+
+enum ChainType {
+	Relay = 'Relay',
+	Parachain = 'Parachain',
+}
+
+const SCALE = new BN(10000);
+
+export class CoretimeService extends AbstractService {
```

Comment:
I checked locally, and all the types are correctly generated for all the storage queries so there isn't a need to typecast here. At least for me I had no errors building when removing all the typecasts.